### PR TITLE
feat: add update and rollback commands, and cli new version update core plugin 

### PR DESCRIPTION
**Pull request by shazron**
## Description

The `update` command will update all installed plugins by checking for any updated versions in the npm registry (with --interactive mode available). Note that this will not check for any updates for any non @adobe namespaced core plugins, for compatibility reasons.

The `rollback` command will uninstall all user-installed plugins (with --interactive mode available).

Through a hook via the `@oclif/plugin-warn-if-update-available` plugin, npm is checked whether there is a newer version of the CLI available, and it will warn the user. This is shown at every command (can't turn it off) but it has a timeout of 7 days (so after 7 days of warnings, it won't prompt the user anymore).

## Motivation and Context

1. Update of installed plugins (including core plugins) without a CLI reinstall
2. Reminder to user that a newer cli version is available

## How Has This Been Tested?

npm test
manual tests

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [X] My code follows the code style of this project.
- [X] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/commands/update.js [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+
+    return plugins
+  }
+
+  /**
+   * Command entry point
+   */
+  async run () {
+    const { flags } = this.parse(UpdateCommand)
+    const spinner = ora()
```
Comment (purplecabbage) [inline src/commands/update.js:156]: Why not use `cli-ux.action.start/stop` and not add a dependency on `ora`?

File: src/commands/update.js [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+
+  /**
+   * Process the plugins, determine if they need updates or warnings.
+   *
+   * @param {object} cliRoot the root folder of the cli
+   * @param {Array<string>} corePlugins package names of the core plugins
+   * @param {Array<InstalledPlugin>} installedPlugins
+   * @returns {Array<ToUpdatePlugin}
+   */
+  async __processPlugins () {
```
Comment (purplecabbage) [inline src/commands/update.js:104]: nit: At some point some of us started putting double __methodPrefix to signify that something is private or not exported.  Personally, I don't really like this practice.
https://github.com/airbnb/javascript#naming--leading-underscore

File: src/commands/update.js [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+        corePlugins.includes(plugin.name) &&
+        !(plugin.name.startsWith('@adobe/'))
+      )
+    )
+
+    for (const plugin of installedPlugins) {
+      const { type, name, version: currentVersion } = plugin
+      const latestVersion = await getNpmLatestVersion(name)
+      let coreVersion = (type === 'core') ? currentVersion : null
+      const needsUpdate = (latestVersion > currentVersion)
```
Comment (purplecabbage) [inline src/commands/update.js:122]: I think we probably need to go full semver here and not just text compare, there are numerous edge cases which I don't see covered by tests. ie. `v1.0.0` or `2.8.3-lts`
https://www.npmjs.com/package/semver

File: src/commands/update.js (hunk 1 above)
Comment (shazron) [reply src/commands/update.js:156]: actually `cli-ux` is not even a dependency of the base cli (check package.json). It just so happens to be a dependency of a dependency (I'm thinking runtime plugin).  Also, I needed a spinner that "goes away" after use, cli-ux's spinner does not do that.

File: src/commands/update.js (hunk 2 above)
Comment (shazron) [reply src/commands/update.js:104]: Initially I wanted the functions into a the helper module. But since these functions heavily use base command functionality (this.config, etc), I kept it here. Until we have private functions in EcmaScript classes, this visual convention would have to do. The only public contract we have with devs is the run() function, nothing else.

File: src/commands/update.js (hunk 3 above)
Comment (shazron) [reply src/commands/update.js:122]: good catch, i'll fix

File: src/commands/update.js (hunk 1 above)
Comment (shazron) [reply src/commands/update.js:156]: Looks like the `discover` command uses `cli-ux` as well, which is not a dependency as I mentioned. In any case, we need to add `cli-ux` as a dependency here because of `cli.table`.

Comment (purplecabbage) [reply src/commands/update.js:156]: how did it ever work?

Comment (shazron) [reply src/commands/update.js:156]: i mentioned above "dependency of a dependency had cli-ux". so it was in node_modules

Comment (shazron) [reply src/commands/update.js:156]: there is a eslint plugin that would have detected this https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/no-extraneous-dependencies.md

Comment (shazron) [reply src/commands/update.js:156]: forgot to mention (See the Slack thread), `aio clear` is now `aio rollback`. I reflected it in the issue description and title when I finalized the PR yesterday.

File: src/commands/update.js (hunk 3 above)
Comment (shazron) [reply src/commands/update.js]: @purplecabbage PR updated
